subprojects/gradle/code-ownership/src/main/kotlin/com/avito/android/CheckProjectDependenciesOwnershipTask.kt

```diff
@@ -29,42 +28,39 @@ abstract class CheckProjectDependenciesOwnershipTask : DefaultTask() {
     }
 
     private fun checkPrivateDependenciesRule() {
-        projectDependencies().forEach {
-            val codeOwnershipExtensionTo = it.module.project.extensions.ownership
-            if (codeOwnershipExtensionTo.visibility == Visibility.PRIVATE) {
-                throwInvalidDependencyOnPrivateVisibleModuleException(
-                    project.path,
-                    it.module.project.path,
-                    codeOwnershipExtensionTo.team
-                )
-            }
-        }
-    }
-
-    private fun checkUnitDependencyRuleDependency() {
-        val codeOwnershipExtensionFrom = project.extensions.ownership
-        projectDependencies().forEach {
-            val codeOwnershipExtensionTo = it.module.project.extensions.ownership
-            if (codeOwnershipExtensionTo.visibility == Visibility.TEAM && codeOwnershipExtensionFrom.team != codeOwnershipExtensionTo.team) {
-                if (codeOwnershipExtensionFrom.allowedDependencies.contains(it.path)) {
-                    logger.warn("WARNING: ${project.path} has forbidden dependency on ${it.path}, but was allowed explicitly in configuration")
-                } else {
-                    throwInvalidDependencyOnUnitVisibleModuleException(
+        project.internalModule
+            .implementationConfiguration
+            .dependencies.forEach {
+                val codeOwnershipExtensionTo = it.module.project.extensions.ownership
+                if (codeOwnershipExtensionTo.visibility == Visibility.PRIVATE) {
+                    throwInvalidDependencyOnPrivateVisibleModuleException(
                         project.path,
                         it.module.project.path,
                         codeOwnershipExtensionTo.team
                     )
                 }
             }
-        }
     }
 
-    private fun projectDependencies() = project.internalModule
-        .implementationConfiguration
-        .dependencies
-        .filterNot {
-            it.module.project.pluginManager.hasPlugin("java-platform")
```

Comment:
Sometimes this check passes and we've got:
```
at com.avito.android.CodeOwnershipExtensionKt.getOwnership(CodeOwnershipExtension.kt:36)
at com.avito.android.CheckProjectDependenciesOwnershipTask.checkPrivateDependenciesRule(CheckProjectDependenciesOwnershipTask.kt:33)
```
I didn't found reasons. 
The task is not cacheable, not UP-TO-DATE. We check the plugin in the execution phase.
All my attempts to test, debug, or log additional info were failed - bug become unreproducible.
It looks like different values due to some race but I have no proofs.

I've decided to avoid this check and make ownership extension optional. The default state seems appropriate and more robust in similar cases with other plugins.